### DISCUSSION TITLE
Moving the CMS State to Vuex

### DISCUSSION THREAD

**Discussion by NiklasLimberg** [shopware/platform#3156] [RFC / Proposal]
Effort: high
Priority: high

### Description:
The CMS Pageeditor in the Administration uses a deeply nested page object, which is then being passed down to the sections, blocks and then slots.

In order to preserve the one way data flow in the current data structure changes need to be propagated up the component chain to the base of the CMS editor to then be passed back up as props. 

This round trip could be prevented  by moving the entire state into the store in a flattened form. So that changes to blocks and slots could be made by a simple store mutation to avoid the round trip and to ensure data consistency.

To illustrate this nesting further, here is the path to the content of a text element: `page.sections[0].blocks[0].slots[0].config.content.value`.
### Benefits:
-  Prevents reactivity pitfalls
-  Enables incremental updates of Blocks and Section via the API
-  Centralises CMS State and Business Logic for common Operations, like moving blocks or changing block state
-  Reduces size of `Props` being passed around
-  Potentially improves performance
-  Potentially easier integration of the [admin-extension-sdk](https://github.com/shopware/admin-extension-sdk) 
### Break strategy
Implement a polyfill in `sw-cms-section/index.js` to map the state in the store to the usual block and section props and event listeners. Grace period till 6.6, then removal of the polyfill and full adoption of the store.

**Comment by lasomethingsomething** [comment]
Thank you for your proposal, @NiklasLimberg. :) Due to lack of activity around this in recent years and the likelihood that this won't change anytime soon, we'll close this proposal for now.
